framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2955,7 +2955,21 @@ private AnnotationMirror getDeclAnnotation(
 
         Set<AnnotationMirror> results = AnnotationUtils.createAnnotationSet();
         // Retrieving the annotations from the element.
-        results.addAll(elements.getAllAnnotationMirrors(elt));
+        List<? extends AnnotationMirror> fromEle = elements.getAllAnnotationMirrors(elt);
+        for (AnnotationMirror annotation : fromEle) {
+            try {
```

Comment:
The ErrorReporter has a reference to the ErrorHandler and the ErrorHandler is a checker.
So we could introduce a handler.issueWarning and have that output as a warning.